p2p/host/peerstore/pstoremem/addr_book.go

```diff
@@ -238,19 +238,21 @@ func (mab *memoryAddrBook) addAddrsUnlocked(s *addrSegment, p peer.ID, addrs []m
 
 	exp := mab.clock.Now().Add(ttl)
 	for _, addr := range addrs {
-		if addr == nil {
+		// Remove suffix of /p2p/peer-id from address
+		taddr, _ := peer.SplitAddr(addr)
```

Comment:
same comment as above about the peer id check. Also I think it's okay to shadow the `addr variable` with the new multiaddr.

Reply:
This should be fixed